### PR TITLE
Feature/search

### PR DESCRIPTION
**Pull request by jgunnink**
Added better date matching for UX and removed column sorting for users. Completed with tests.

### CODE REVIEW

File: spec/models/user/dashboard_search_spec.rb [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    context 'when a date is entered for users created before AND after' do
+      let(:params) { {date_start: "01/11/2015", date_end: "30/11/2015"} }
+
+      it { should include(FactoryGirl.create(:user, created_at: "02/11/2015")) }
+      it { should_not include(FactoryGirl.create(:user, created_at: "30/10/2015")) }
+      it { should_not include(FactoryGirl.create(:user, created_at: "01/12/2015")) }
+    end
+
+    context 'all' do
```
Comment (colynprater) [inline spec/models/user/dashboard_search_spec.rb]: 'when all params are passed' might be more informative than 'all'

Comment (jgunnink) [inline spec/models/user/dashboard_search_spec.rb]: Good point. I think I'll update to `"when all form fields are filled out with valid data"`

File: spec/models/user/dashboard_search_spec.rb [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      let(:params)       { {user_role: "admin"} }
+
+      it { should include role_user }
+      it { should_not include user_that_doesnt_match }
+    end
+
+    context 'when an email is present' do
+      let(:params) { {email: "xyz" } }
+
+      it { should include(FactoryGirl.create(:user, email: "abc@xyz.com")) }
```
Comment (colynprater) [inline spec/models/user/dashboard_search_spec.rb]: good move testing users with emails like the one being passed are returned

File: spec/models/user/dashboard_search_spec.rb [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+          site_name: "Hobbit Land",
+          state_name: "Middle Earth",
+          suburb_name: "The Shire",
+          user_role: :admin
+        }
+      end
+
+      let(:address)     { FactoryGirl.create(:address, suburb: "The Shire", city: "Place", state: state) }
+      let(:user)        { FactoryGirl.create(:user, created_at: "2015-11-15", email: "abc@example.com", role: :admin, site: site) }
+      let(:site)        { FactoryGirl.create(:site, name: "Hobbit Land", address: address) }
```
Comment (colynprater) [inline spec/models/user/dashboard_search_spec.rb]: Hobbiton*

File: spec/models/user/dashboard_search_spec.rb [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      let(:state)       { FactoryGirl.create(:state, name: "Middle Earth") }
+
+      it { should include(user) }
+      it { should_not include user_that_doesnt_match }
```
Comment (colynprater) [inline spec/models/user/dashboard_search_spec.rb]: user_that_doesnt_match reads like english in the assertion. is good

**Pull request merged**